bin/reth/src/commands/stage/unwind.rs

```diff
@@ -59,25 +61,35 @@ impl Command {
             eyre::bail!("Database {db_path:?} does not exist.")
         }
 
-        let db =
-            open_db(db_path.as_ref(), DatabaseArguments::default().log_level(self.db.log_level))?;
+        let db = Arc::new(open_db(
+            db_path.as_ref(),
+            DatabaseArguments::default().log_level(self.db.log_level),
+        )?);
 
         let range = self.command.unwind_range(&db)?;
 
         if *range.start() == 0 {
             eyre::bail!("Cannot unwind genesis block")
         }
 
-        let factory = ProviderFactory::new(&db, self.chain.clone(), data_dir.static_files_path())?;
-        let provider = factory.provider_rw()?;
+        let provider_factory =
+            ProviderFactory::new(db.clone(), self.chain.clone(), data_dir.static_files_path())?;
 
-        let blocks_and_execution = provider
-            .take_block_and_execution_range(&self.chain, range)
-            .map_err(|err| eyre::eyre!("Transaction error on unwind: {err}"))?;
+        let pipeline_builder = Pipeline::builder();
 
-        provider.commit()?;
+        let mut pipeline = pipeline_builder.build(
+            provider_factory.clone(),
+            StaticFileProducer::new(
+                provider_factory.clone(),
+                provider_factory.static_file_provider(),
+                PruneModes::default(),
+            ),
+        );
 
-        println!("Unwound {} blocks", blocks_and_execution.len());
+        pipeline.unwind(*range.end(), None)?;
```

Comment:
i think it should rather be the `*range.start()`

Reply:
fixed